include/arch/arm/aarch32/cortex_m/nvic.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2020, Linaro Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_ARM_AARCH32_CORTEX_M_NVIC_H_
+#define ZEPHYR_INCLUDE_ARCH_ARM_AARCH32_CORTEX_M_NVIC_H_
+
+#ifdef CONFIG_ARMV6_M_ARMV8_M_BASELINE
+#define NVIC_NODEID DT_INST(0, arm_v6m_nvic)
+/* The order here is on purpose since v8m SoC's can define
```

Comment:
The comment is legitimate, but then you should do the right order here:
```
#if v8m_base || v8m_main
use v8m_nvic
#elif v6_m_v8_m_base
use v6m_nvic
#elif v7m_v8m_main
use v7m_nvic
```

Reply:
actually, the comment needs to add the v6_m compound macro as well:

Reply:
@galak I think you need to have:

```
#if defined(CONFIG_ARMV8_M_BASELINE) || defined(CONFIG_ARMV8_M_MAINLINE)
/* The order here is on purpose since ARMv8-M SoCs may define
 * CONFIG_ARMV6_M_ARMV8_M_BASELINE or CONFIG_ARMV7_M_ARMV8_M_MAINLINE
 *  so we want to check for ARMv8-M first.
 */
#define MPU_NODEID DT_INST(0, arm_v8m_nvic)
#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
#define MPU_NODEID DT_INST(0, arm_v7m_nvic)
#elif defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
#define MPU_NODEID DT_INST(0, arm_v6m_nvic)
#endif
```